infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -708,4 +728,47 @@ public void shutdown() {
     private String getLocalHost() {
         return NodeLocator.parseString(serverContext.getLocalEndpoint()).getHost();
     }
+
+    private LockClient createLockClient(LockStateType initState) throws Exception {
+
+        LockConfig lockConfig = createLockConfig(serverContext.getPluginConfigFilePath());
+        LockClient lockClient =
+                new LockClient(logReplicationNodeId,
+                        lockConfig, runtime.get());
+
+        LockListener lockListener =
+                new LogReplicationLockListener(this);
+
+        LockDataTypes.LockId lockId = LockDataTypes.LockId.newBuilder()
+                .setLockGroup(lockConfig.getLockGroup())
+                .setLockName(lockConfig.getLockName())
+                .build();
+
+        lockClient.getLocks().computeIfAbsent(
+                lockId,
+                key -> new Lock(lockId, lockListener, lockClient.getClientContext(), lockConfig,
+                        initState));
+
+        return lockClient;
+    }
+
+
+    @VisibleForTesting
+    public void deregisterToLogReplicationLock() throws Exception {
+        LockClient lockClient = createLockClient(LockStateType.HAS_LEASE);
```

Comment:
this will create a new lock every time, right?  can we use the same lock keep releasing/acquiring it?

Reply:
With how the locking is designed, it gets way too complex if we reuse the instance. It is much easier to disable the locking all together, put it in some initial state and then invoke the event that forces the transition. It has a @VisibleForTesting annotation, so this method will not be used in prod anyways.